app/src/main/java/ch/epfl/sdp/GameArea.java

```diff
@@ -0,0 +1,59 @@
+package ch.epfl.sdp;
+
+import java.util.Random;
+
+import static java.lang.Math.cos;
+import static java.lang.Math.sin;
+import static java.lang.Math.sqrt;
+import static java.lang.Math.toRadians;
+
+public class GameArea {
+    private double radius;
+    private GeoPoint center;
+
+    public GameArea(double radius, GeoPoint center) {
+        this.radius = radius;
+        this.center = center;
+    }
+
+    public GameArea shrink(double factor) {
+        if (factor < 0 || factor > 1) {
+            return null;
+        }
+        Random random = new Random();
+        double u = random.nextDouble();
+        double v = random.nextDouble();
+        double w = ((1.0-factor)*radius/111300.0) * sqrt(u);
+        double t = 2 * Math.PI * v;
+        double x = w * cos(t);
+        double y = w * sin(t);
+        double xPrime = x / cos(toRadians(y));

```

Comment:
Actually, I do not fully understand this formula, is it possible to have more meaningful variable names or add some comments?

Reply:
I will try, but since it is some maths (basically we are looking for a random point that is inside a circle, knowing the fact we are on earth, so it's not flat)

Reply:
or you can use javadoc to describe this?

Reply:
I will try to do this